### PR TITLE
Fixes magboot fuckery

### LINKED ISSUE #26740: Holding magboots in pocket still provide you magpulse trait
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

 515.1643

### Issue Description

Toggle magboot, have magboot profit just holiding it in hand or inventory.

### What did you expect to happen?

I toggle magboot and it not working untill it weared properly

### What happened instead?

I toggle magboot, put it in backpack and still have all magboot advantages 

### Why is this bad/What are the consequences?

this is obviously not how it must work

### Steps to reproduce the issue.

Find any magboot
Toggle it in any means
While holding it in backpack, you have gravity immune and noslip trait while wearing jump boots or modsuit.

### When did the problem start happening?

dont know, notice 2-3 days ago

### Extra information

_No response_

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

Fixes magboot fuckery. Specifically fixes #26740 as an alternate PR to #26788 with permission of the author.
Also renames the internal magpulse trait from "magnetificent" to "magpulse" (its funny but without codediving its harder to tell what it is)

## Why It's Good For The Game

Fixes a bug

## Testing

Picked up the boots, turned them on (magpulse not active)
Put the boots on my feet (magpulse active)
took the boots off my feet (magpulse not active)
turned the magboots off (magpulse not active)
put the boots on my feet (magpulse not active)
turned the magboots on (magpulse active)
took the magboots off (magpulse not active)

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

:cl:
fix: Magboots only work when equipped
/:cl:

### CODE REVIEW

**Comment by TravisAngeI** [comment]
> renames the internal magpulse trait from "magnetificent" to "magpulse"

GAME RUINED...

File: code/modules/clothing/shoes/magboots.dm [hunk 1]
```diff
@@ -26,45 +26,47 @@
...
 		return
-	if(magpulse)
-		ADD_TRAIT(user, TRAIT_MAGPULSE, "magboots")
+	if(magpulse && !removing)
+		ADD_TRAIT(user, TRAIT_MAGPULSE, "magboots[UID()]")
 		return
 	if(HAS_TRAIT(user, TRAIT_MAGPULSE)) // User has trait and the magboots were turned off, remove trait
-		REMOVE_TRAIT(user, TRAIT_MAGPULSE, "magboots")
+		REMOVE_TRAIT(user, TRAIT_MAGPULSE, "magboots[UID()]")
+		// REMOVE_TRAIT(H, TRAIT_NOSLIP, UID()) // have to shove this in because
```
Comment (lewcc) [inline code/modules/clothing/shoes/magboots.dm:69]: commented out code

Comment (Contrabang) [inline code/modules/clothing/shoes/magboots.dm:69]:
```suggestion
```

File: code/__HELPERS/trait_helpers.dm [hunk 2]
```diff
@@ -225,7 +225,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_IPC_JOINTS_SEALED "ipc_joints_sealed" // The IPC's limbs will not pop off bar sharp damage (aka like a human), but will take slightly more stamina damage
 #define TRAIT_HAS_GPS "has_gps" // used for /Stat
 #define TRAIT_CAN_VIEW_HEALTH "can_view_health" // Also used for /Stat
-#define TRAIT_MAGPULSE "magnetificent" // Used for anything that is magboot related
+#define TRAIT_MAGPULSE "magpulse" // Used for anything that is magboot related
```
Comment (DGamerL) [inline code/__HELPERS/trait_helpers.dm:228]: I understand why this is done, it still pains me

File: code/modules/clothing/shoes/magboots.dm [hunk 3]
```diff
@@ -26,45 +26,46 @@
...
 	if(!user)
 		return
-	if(magpulse)
-		ADD_TRAIT(user, TRAIT_MAGPULSE, "magboots")
+	if(magpulse && !removing)
+		ADD_TRAIT(user, TRAIT_MAGPULSE, "magboots[UID()]")
 		return
 	if(HAS_TRAIT(user, TRAIT_MAGPULSE)) // User has trait and the magboots were turned off, remove trait
-		REMOVE_TRAIT(user, TRAIT_MAGPULSE, "magboots")
+		REMOVE_TRAIT(user, TRAIT_MAGPULSE, "magboots[UID()]")
```
Comment (DGamerL) [inline code/modules/clothing/shoes/magboots.dm:68]: I think changing these both to `UNIQUE_TRAIT_SOURCE` could be a good idea

Comment (Contrabang) [reply code/modules/clothing/shoes/magboots.dm:68]: `#define UNIQUE_TRAIT_SOURCE(target) "unique_source_[UID(target)]"`
This define doesn't even work properly, and I think its better to have more information where its actually coming from so people debugging don't have to find the UID to know what is providing the trait

Comment (DGamerL) [reply code/modules/clothing/shoes/magboots.dm:68]: PR is up to fix it, we probably don't need to enforce it to always be used but a UID should always be used

**Review by DGamerL — APPROVED** [review]

**Pull request merged**